PolyEngine/Engine/Src/TransformComponent.cpp

```diff
@@ -0,0 +1,32 @@
+#include "TransformComponent.hpp"
+
+using namespace Poly;
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalTranslation(Vector position) {
+	LocalTranslation = position;
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalRotation(EulerAngles angles) { 
+	LocalRotation = Quaternion(angles); 
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalScale(Vector scale) {
+	LocalScale = scale;
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+Matrix TransformComponent::GetLocalTransformationMatrix() {
+	Matrix rotation = LocalRotation.ToRotationMatrix();
+	Matrix translation;
+	Matrix scale;
+	translation.SetTranslation(LocalTranslation);
+	scale.SetScale(LocalScale);
+	LocalTransform = rotation * translation * scale;
```

Comment:
Please change the order of the multiplication. It should be translation * rotation * scale.
Reference: https://gamedev.stackexchange.com/questions/16719/what-is-the-correct-order-to-multiply-scale-rotation-and-translation-matrices-f